rts/System/SpringDampers.cpp

```diff
@@ -0,0 +1,105 @@
+/* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html
+
+	Spring Damper Functions are MIT Licensed Copyright 2021 Daniel Holden
+	https://github.com/orangeduck/Spring-It-On
+	https://theorangeduck.com/page/spring-roll-call
+*/
+
+#include "System/SpringDampers.h"
+
+float halflife_to_damping(float halflife, float eps)
+{
+	return (4.0f * 0.69314718056f) / (halflife + eps);
+}
+
+float fast_negexp(float x)
+{
+	return 1.0f / (1.0f + x + 0.48f*x*x + 0.235f*x*x*x);
+}
+
+float spring_damper_damping(float halflife)
+{
+	return halflife_to_damping(halflife) / 2.0f;
```

Comment:
```suggestion
	static constepxr float why_halve? = 0.5f;
	return halflife_to_damping(halflife) * well_named_constant;
```

Reply:
These constants are all related to the derivation of stiffness and damping inputs to a single halflife value. The math is a bit too complicated for me so you'll have to read the guy's blogpost if you want more info. But my take away from it is that we can have a single input value that gives it a more or less "springiness".